lib/vm.js

```diff
@@ -233,10 +234,35 @@ function createContext(contextObject = {}, options = {}) {
     validateBoolean(wasm, 'options.codeGeneration.wasm');
   }
 
-  makeContext(contextObject, name, origin, strings, wasm);
+  return { name, origin, strings, wasm };
+}
+
+function createContext(contextObject = {}, options = {}) {
+  if (isContext(contextObject)) {
+    return contextObject;
+  }
+
+  const { name, origin, strings, wasm } = validateContextOptions(options);
+
+  makeContext(contextObject, name, origin, strings, wasm, undefined);
   return contextObject;
 }
 
+class Context {
+  #global;
+
+  constructor(options = {}) {
+    emitExperimentalWarning('vm.Context');
+
+    const { name, origin, strings, wasm } = validateContextOptions(options);
+    this.#global = makeContext(undefined, name, origin, strings, wasm, this);
+  }
+
+  get global() {
```

Comment:
This actually refers to the global proxy: <https://github.com/nodejs/node/pull/30709/files#r392627951>
```suggestion
  get globalProxy() {
```